packages/flutter/test/widgets/editable_text_shortcuts_test.dart

```diff
@@ -2365,5 +2382,522 @@ void main() {
+        const SingleActivator pageDown = SingleActivator(LogicalKeyboardKey.pageDown);
+
+        controller.text = longText;
+        controller.selection = const TextSelection.collapsed(offset: 0);
+
+        await tester.pumpWidget(buildEditableText(style: const TextStyle(fontSize: 12)));
+        await tester.pumpAndSettle();
+
+        final double initialScrollOffset = scrollController.offset;
+
+        expect(initialScrollOffset, 0);
+        await sendKeyCombination(tester, pageDown);
+        await tester.pump();
+
+        expect(controller.text, longText);
+        expect(
+          controller.selection,
+          const TextSelection.collapsed(offset: 0), // selection stays the same
+          reason: pageDown.toString(),
+        );
+
+        // default page up/down scroll increment is 80% of viewport dimension.
+        final double newOffset = initialScrollOffset + (0.8 * scrollController.position.viewportDimension);
+
+        expect(scrollController.offset, newOffset);
+      }, variant: macOSOnly);
+
+      testWidgetsWithLeakTracking('pageDown + shift scrolls downwards and modifies selection', (WidgetTester tester) async {
+        const SingleActivator pageDown = SingleActivator(LogicalKeyboardKey.pageDown, shift: true);
+
+        controller.text = longText;
+        controller.selection = const TextSelection.collapsed(offset: 0);
+
+        await tester.pumpWidget(buildEditableText(style: const TextStyle(fontSize: 12)));
+        await tester.pumpAndSettle();
+
+        final double initialScrollOffset = scrollController.offset;
+
+        expect(initialScrollOffset, 0);
+        await sendKeyCombination(tester, pageDown);
+        await tester.pump();
+
+        expect(controller.text, longText);
+        expect(
+          controller.selection,
+          const TextSelection(baseOffset: 0, extentOffset: 238),
+          reason: pageDown.toString(),
+        );
+
+        expect(scrollController.offset, greaterThan(initialScrollOffset));
+      }, variant: macOSOnly);
+
+      testWidgetsWithLeakTracking('end scrolls to the end of the text field', (WidgetTester tester) async {
+        const SingleActivator end = SingleActivator(LogicalKeyboardKey.end);
+
+        controller.text = longText;
+        controller.selection = const TextSelection.collapsed(offset: 0);
+
+        await tester.pumpWidget(buildEditableText(style: const TextStyle(fontSize: 12)));
+        await tester.pumpAndSettle();
+
+        final double initialScrollOffset = scrollController.offset;
+
+        expect(initialScrollOffset, 0);
+        await sendKeyCombination(tester, end);
+        await tester.pump();
+
+        expect(controller.text, longText);
+        expect(
+          controller.selection,
+          const TextSelection.collapsed(offset: 0), // selection stays the same
+          reason: end.toString(),
+        );
+
+        // scrolls to end
```

Comment:
nit: consider ending comment with a period here and in other places.

Reply:
fixed